apps/notifications/src/standalone/index.jsx

```diff
@@ -70,10 +70,59 @@ const customMiddleware = {
 	],
 };
 
-const render = ( wpcom ) => {
-	document.body.classList.add( 'font-smoothing-antialiased' );
+const NotesWrapper = ( { wpcom } ) => {
+	const [ isShowing, setIsShowing ] = useState( false );
+	const [ isVisible, setIsVisible ] = useState( document.visibilityState === 'visible' );
 
-	ReactDOM.render(
+	debug( 'wrapper state update', { isShowing, isVisible } );
+
+	const refresh = () =>
+		setIsVisible( ( prevIsVisible ) => {
+			store.dispatch( { type: 'APP_REFRESH_NOTES', isVisible: prevIsVisible } );
+			return prevIsVisible;
+		} );
+	const reset = () => store.dispatch( { type: 'SELECT_NOTE', noteId: null } );
+
+	const handleMessages = ( { action, hidden, showing } ) => {
+		debug( 'message received', {
+			action,
+			hidden,
+			showing,
+		} );
+
+		if ( 'togglePanel' === action ) {
+			setIsShowing( ( prevIsShowing ) => {
+				if ( prevIsShowing && ! showing ) {
+					reset();
+				}
+				return showing;
+			} );
+			refresh();
+		}
+
+		if ( 'toggleVisibility' === action ) {
+			setIsVisible( ! hidden );
+			refresh();
+		}
+
+		if ( 'refreshNotes' === action ) {
+			refreshNotes();
+		}
+	};
+
+	useEffect( () => {
+		document.addEventListener( 'visibilitychange', refresh );
+
+		window.addEventListener( 'message', receiveMessage( handleMessages ) );
+
+		// Ensure empty dependency array so we only add these listeners once. The functionality of
```

Comment:
Thanks for adding this comment to explain this, I don't 1000% understand but it seems reasonable.

Reply:
>  I don't 1000% understand

Happy to try and explain more.

An empty dependency array will mean useEffect only runs on mount. No dependency array means it will run every render.  While we want the empty dependency behavior, the linter gets mad because it sees the function relying on values like `handleMessages` and suggests adding those to the dependency array. 

IF we add those to the dependency array, it means the useEffect will rerun every time they change (adding more listeners to the dom in this case). In their current state that would be every render since those functions are simply defined in the functional component. We could memoize them with `useCallback` or something similar so they do get compared as the same value and as non changing and that would stop the component from updating unnecessarily in that case. But thats just using some extra resources as a work around to a linter rule we could just comment out. 😅   

The current functions used in that useEffect don't have any changing dependencies themselves, and will always have the same functionality. So we have no need for them to ever be updated or changed once they are already on the DOM anyways.